Description: Fix broken unit tests for get_ilo_object
 First, the tested function signature was wrong. We didn't catch it in gate,
 as we mock proliantutils, but it does break e.g. Debian package build.
 .
 Second, the arguments override was not actually working. We didn't catch
 it in gate, because the new values were the same as the defaults.
Author: Dmitry Tantsur <divius.inside@gmail.com>
Date: Wed, 21 Sep 2016 13:45:21 +0000 (+0200)
X-Git-Url: https://review.openstack.org/gitweb?p=openstack%2Fironic.git;a=commitdiff_plain;h=87327803772ef7c80f7981d32851b90253d5c655
Bug-Ubuntu: #1626089
Change-Id: I2e4899e368b0b882dcd59bf33fdca98f47e5b405
Origin: upstream, https://review.openstack.org/374161
Last-Update: 2016-09-21

diff --git a/ironic/tests/unit/drivers/modules/ilo/test_common.py b/ironic/tests/unit/drivers/modules/ilo/test_common.py
index d61a7a0..c1bf3cb 100644
--- a/ironic/tests/unit/drivers/modules/ilo/test_common.py
+++ b/ironic/tests/unit/drivers/modules/ilo/test_common.py
@@ -154,9 +154,10 @@ class IloCommonMethodsTestCase(db_base.DbTestCase):
     @mock.patch.object(ilo_client, 'IloClient', spec_set=True,
                        autospec=True)
     def _test_get_ilo_object(self, ilo_client_mock, isFile_mock, ca_file=None):
-        self.info['client_timeout'] = 60
-        self.info['client_port'] = 443
+        self.info['client_timeout'] = 600
+        self.info['client_port'] = 4433
         self.info['ca_file'] = ca_file
+        self.node.driver_info = self.info
         ilo_client_mock.return_value = 'ilo_object'
         returned_ilo_object = ilo_common.get_ilo_object(self.node)
         ilo_client_mock.assert_called_with(
@@ -164,7 +165,8 @@ class IloCommonMethodsTestCase(db_base.DbTestCase):
             self.info['ilo_username'],
             self.info['ilo_password'],
             self.info['client_timeout'],
-            self.info['client_port'])
+            self.info['client_port'],
+            cacert=self.info['ca_file'])
         self.assertEqual('ilo_object', returned_ilo_object)
 
     def test_get_ilo_object_cafile(self):
